pkg/ccl/backupccl/backup_test.go

```diff
@@ -1443,6 +1443,252 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
+		{
+			// The full backup will only contain backup data written after backupStartTime,
+			// t0, so we expect any recorded checkpoints to be for regular backup
+			// spans.
+			name:      "full-backup",
+			backupCmd: fullBackupDetachedCmd,
+			preBackupWorkload: func(ctx context.Context, t *testing.T, sqlDB *sqlutils.SQLRunner) {
+				createTable(sqlDB, "t1")
+			},
+		},
+		{
+			// The incremental backup will only contain backup data written after
+			// backupStartTime, so we expect any recorded checkpoints to be for
+			// regular backup spans.  NB: the backup will send two export requests:
+			// one from [t0,BackupStartTime) and another for
+			// [backupStartTime,BackupEndTime), as the created table is technically an
+			// introduced span. But, only the second request will yield any data and
+			// affect checkpointing, and the create table operation is mvcc and ran
+			// after the full backup ended.
+			name:      "inc-backup",
+			backupCmd: incBackupDetachedCmd,
+			preBackupWorkload: func(ctx context.Context, t *testing.T, sqlDB *sqlutils.SQLRunner) {
+				// Required to add a table for the first backup because of
+				// https://github.com/cockroachdb/cockroach/issues/115773
+				createTable(sqlDB, "t1")
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO '%s'`, localFoo))
+				createTable(sqlDB, "t2")
+			},
+		},
+		{
+			// The incremental backup will _only_ back up actual data from before
+			// backupStartTime, so we expect any recorded checkpoints to be for
+			// introduced spans. NB: the tracked backup will send two export requests:
+			// one from [t0,BackupStartTime) and another for
+			// [backupStartTime,BackupEndTime). Only the first request will yield any
+			// data and thus affect checkpointing, however both requests will ping the
+			// progress channel in a non deterministic order, so it's very hard to
+			// capture the checkpoint manifest created after the non-empty response
+			// returns.
+			name:      "introduce-spans",
+			backupCmd: incBackupDetachedCmd,
+			preBackupWorkload: func(ctx context.Context, t *testing.T, sqlDB *sqlutils.SQLRunner) {
+				// Runs a workload such that the next backup of the
+				// provided table will back up all data from the table since t0.
+				tableName := "t1"
+				createTable(sqlDB, tableName)
+				sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_publishing_descriptors'`)
+
+				// t0: Backup a table that we'll then restore with pause
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO '%s'`, localFoo))
+				sqlDB.Exec(t, fmt.Sprintf(`DROP TABLE %s`, tableName))
+				var restoreJobID int
+				sqlDB.QueryRow(t, fmt.Sprintf(`RESTORE TABLE %s FROM LATEST IN '%s' with detached`, tableName, localFoo)).Scan(&restoreJobID)
+				jobutils.WaitForJobToPause(t, sqlDB, jobspb.JobID(restoreJobID))
+
+				// t1: run an inc backup while restore is paused to ensure the introduced span contains data in the subseq
```

Comment:
in the subseq -> in the subsequent backup.

Reply:
Done.